### PR TITLE
feat: Add compact database tool

### PR DESCRIPTION
**Pull request by jbajic**
Add a database tool for running SST file compactions. This is useful when changing block size parameters in order to have an optimized layout of data in SST files.

### CODE REVIEW

File: tools/database/README.md [hunk 1]
```diff
@@ -62,6 +62,18 @@ Then you can call
 `neard database change-db-kind --new-kind Cold change-hot`.
 Notice that even though in your mind this db is cold, in your config this db hot, so you have to pass `change-hot`.
 
+## Compact database
+
+Run manual compaction on the SST files. After running this the performance of database will increase significantly.
```
Comment (nikurt) [inline tools/database/README.md:67]: > After running this the performance will increase significantly

What happens if I run it 10 times in a row? Will all operations become instant? :)

File: tools/database/src/compact.rs [hunk 2]
```diff
@@ -0,0 +1,17 @@
+use clap::Parser;
+use near_store::db::Database;
+use std::path::PathBuf;
+
```
Comment (nikurt) [inline tools/database/src/compact.rs:4]: Please don't split imports into blocks: https://github.com/near/nearcore/blob/master/docs/practices/style.md#import-blocks

**Review by nikurt — APPROVED** [review]

File: tools/database/README.md (hunk 1 above)
Comment (jbajic) [reply tools/database/README.md:67]: Not really, from my testing with it, first run takes around 2h and subsequent run it is about 10min, I think it goes through every block of SST file

File: core/store/src/db/rocksdb.rs [hunk 3]
```diff
@@ -377,6 +378,7 @@ impl Database for RocksDB {
     fn compact(&self) -> io::Result<()> {
         let none = Option::<&[u8]>::None;
         for col in DBCol::iter() {
+            info!(target: "db", column = col.to_string(); "Compacted column");
```
[the remark this answers is not in the document]
Comment (jbajic) [reply core/store/src/db/rocksdb.rs:381]: Hmm I don't think this works, this was a typo...

File: core/store/src/db/rocksdb.rs [hunk 4]
```diff
@@ -377,6 +377,7 @@ impl Database for RocksDB {
     fn compact(&self) -> io::Result<()> {
         let none = Option::<&[u8]>::None;
         for col in DBCol::iter() {
+            tracing::info!(target: "db", column = col.to_string(), "Compacted column");
```
Comment (pugachAG) [inline core/store/src/db/rocksdb.rs:380]: nit: use `column = col` or `column = %col`

File: tools/database/README.md [hunk 5]
```diff
@@ -62,6 +62,18 @@ Then you can call
 `neard database change-db-kind --new-kind Cold change-hot`.
 Notice that even though in your mind this db is cold, in your config this db hot, so you have to pass `change-hot`.
 
+## Compact database
+
+Run compaction on the SST files. After running this the performance of database will increase significantly.
```
Comment (pugachAG) [inline tools/database/README.md:67]: suggestion: `After running this the performance of database will increase significantly.` -> `Running this command might increase read performance.`

**Review by pugachAG — APPROVED** [review]

**Pull request merged**